oauthproxy.go

```diff
@@ -375,21 +375,34 @@ func (p *OAuthProxy) MakeCSRFCookie(req *http.Request, value string, expiration
 }
 
 func (p *OAuthProxy) makeCookie(req *http.Request, name string, value string, expiration time.Duration, now time.Time) *http.Cookie {
-	if p.CookieDomain != "" {
-		domain := req.Host
+	cookieDomain := ""
+	host := req.Header.Get("X-Forwarded-Host")
+	if host == "" {
+		host = req.Host
+	}
+
+	for _, domain := range p.CookieDomains {
+		if strings.HasSuffix(host, domain) {
+			cookieDomain = domain
+			break
+		}
+	}
```

Comment:
This could be made into a small method in say the cookies package that returns the cookie domain to use, that could then be used her and in `pkg/cookies/cookies.go`, ideally something that takes the request and a list of cookie domains and returns the string to use, WDYT?

Reply:
Done